src/util/invariant.h

```diff
@@ -242,6 +253,10 @@ inline void invariant_violated_string(
 // changed by a previous method call.
 // "The contract of the previous method call means the following
 //  condition holds".
+//
+// A difference between CHECK_RETURN and POSTCONDITION is that CHECK_RETURN is
+// a statement about what the caller expects from a function, whereas a
+// POSTCONDITION is a statement about guarantees made by the function itself.
```

Comment:
Yes!  This is a subtle point but the key thing, as you have got, is that POSTCONDITION is the callee saying "I will always guarantee that CONDITION holds before I return" while CHECK_RETURN is the caller saying "I will assume that the called function made sure CONDITION holds".  Ideally the POSTCONDITION will imply that all CHECK_RETURNs hold but the separation is to make the separation of responsibilities clear.